src/cargo/util/errors.rs

```diff
@@ -286,6 +286,16 @@ impl From<Box<CargoError>> for CliError {
 }
 
 // =============================================================================
+// NetworkError trait
+
+pub trait NetworkError: Error + Send + 'static {
+    fn maybe_spurious(&self) -> bool { true }
```

Comment:
Did not work out of the box. I need some help deciphering what to do.  I pushed the failing code but here is my errors: https://gist.github.com/sbeckeriv/bd2173d07e98b0e599d6

Thanks.
Becker